packages/opentelemetry-exporter-jaeger/src/jaeger.ts

```diff
@@ -59,47 +53,66 @@ export class JaegerExporter implements SpanExporter {
     resultCallback: (result: ExportResult) => void
   ): void {
     this._logger.debug('Jaeger exporter export');
-    return this._sendSpans(spans, resultCallback);
+    this._sendSpans(spans, resultCallback).catch(err => {
+      this._logger.error(`JaegerExporter failed to export: ${err}`);
+    });
   }
 
   /** Shutdown exporter. */
   shutdown(): void {
-    if (!this._forceFlush) return;
+    if (!this._forceFlushOnShutdown) return;
     // Make an optimistic flush.
     this._flush();
     // Sleeping x seconds before closing the sender's connection to ensure
     // all spans are flushed.
     setTimeout(() => {
       this._sender.close();
-    }, this._flushTimeout);
+    }, this._onShutdownFlushTimeout);
   }
 
   /** Transform spans and sends to Jaeger service. */
-  private _sendSpans(
+  private async _sendSpans(
     spans: ReadableSpan[],
     done?: (result: ExportResult) => void
   ) {
     const thriftSpan = spans.map(span => spanToThrift(span));
     for (const span of thriftSpan) {
-      this._sender.append(span, (numSpans: number, err?: string) => {
-        if (err) {
-          // @todo: decide whether to break out the loop on first error.
-          this._logger.error(`failed to append span: ${err}`);
-          if (done) return done(ExportResult.FAILED_NOT_RETRYABLE);
-        }
-      });
+      try {
+        await this._append(span);
+      } catch (err) {
+        this._logger.error(`failed to append span: ${err}`);
+        if (done) return done(ExportResult.FAILED_NOT_RETRYABLE);
+      }
     }
-    // @todo: We should wait for all the callbacks of the append calls to
-    // complete before it calls done with success.
     this._logger.debug('successful append for : %s', thriftSpan.length);
+
+    // Flush all spans on each export. No-op if span buffer is empty
+    await this._flush();
```

Comment:
Perhaps `flush` should happen whenever `thriftSpan.length > 0` and move above `debug` statement in same `if` loop. WDYT?

Reply:
First thing the sender does is check length and return if zero and this isn't a critical path or tight loop. Can change if you feel strongly though

Reply:
Ok. I was little annoyed by the above debug log, even though nothing was exported it just printed `successful append for : 0` every few secs.

Reply:
Oh i understand, yea that can be fixed

Reply:
https://github.com/open-telemetry/opentelemetry-js/pull/609/commits/755ce9c81d1b980319ffc767a6e6e130127253eb